packages/contracts/ts/utils.ts

```diff
@@ -199,3 +199,26 @@ export const getBlockTimestamp = async (signer: Signer): Promise<number> => {
   const block = await signer.provider?.getBlock("latest");
   return block?.timestamp ?? Math.floor(Date.now() / 1000);
 };
+
+export const getDeployedContractAddressFromContractReceipt: (
```

Comment:
This way we are able to use signer in all our codebase (so in the future we only change the SDK logic) and deploy and retrieve the contract address if the signer is a AA.

This function helps us get the contract address from the event logs

Reply:
make sure to properly comment all functions you add (typedoc style)